src/Model/Resolver/SetBillingAddressOnCart.php

```diff
@@ -84,7 +91,23 @@ public function resolve(
         }
 
         $this->checkCartCheckoutAllowance->execute($cart);
-        $this->setBillingAddressOnCart->execute($context, $cart, $billingAddress);
+        $billingAddressIsSaved = false;
+
+        if (isset($args['input']['same_as_shipping']) && $args['input']['same_as_shipping']) {
+            $customer = $cart->getCustomer();
+            $shippingAddressId = $customer->getDefaultShipping();
+            $billingAddressId = $customer->getDefaultBilling();
+
+            if ($shippingAddressId && !$billingAddressId) {
+                $customer->setDefaultBilling($shippingAddressId);
+                $this->customerRepository->save($customer);
+                $billingAddressIsSaved = true;
+            }
```

Comment:
This seems strange to me, is following case covered here?
- The user shipping address id on cart was not default (rather some other)

Reply:
Fixed it. Now i'm passing chosen address if billing address is set same as shipping.